src/shared/bus-util.h

```diff
@@ -122,6 +122,59 @@ assert_cc(sizeof(mode_t) == sizeof(uint32_t));
+                                                                        \
+                assert(bus);                                            \
+                assert(reply);                                          \
+                                                                        \
+                return sd_bus_message_append(reply, bus_type, val);     \
+        }
+
+#define BUS_DEFINE_PROPERTY_GET(function, bus_type, data_type, get)     \
+        int function(sd_bus *bus,                                       \
+                     const char *path,                                  \
+                     const char *interface,                             \
+                     const char *property,                              \
+                     sd_bus_message *reply,                             \
+                     void *userdata,                                    \
+                     sd_bus_error *error) {                             \
+                                                                        \
+                data_type *data = userdata;                             \
+                                                                        \
+                assert(bus);                                            \
+                assert(reply);                                          \
+                assert(data);                                           \
+                                                                        \
+                return sd_bus_message_append(reply, bus_type,           \
+                                             get(data));                \
+        }
+
+#define BUS_DEFINE_PROPERTY_GET2(function, bus_type, data_type, get1, get2) \
+        int function(sd_bus *bus,                                       \
+                     const char *path,                                  \
+                     const char *interface,                             \
+                     const char *property,                              \
+                     sd_bus_message *reply,                             \
+                     void *userdata,                                    \
+                     sd_bus_error *error) {                             \
+                                                                        \
+                data_type *data = userdata;                             \
+                                                                        \
+                assert(bus);                                            \
+                assert(reply);                                          \
+                assert(data);                                           \
+                                                                        \
+                return sd_bus_message_append(reply, bus_type,           \
+                                             get2(get1(data)));         \
+        }
+
```

Comment:
What about defining
```C
#define ident(x) (x)
#define BUS_DEFINE_PROPERTY_GET(function, bus_type, data_type, get)     \
  BUS_DEFINE_PROPERTY_GET2(function, bus_type, data_type, get1, ident)
#define BUS_DEFINE_PROPERTY_GET_GLOBAL(function, bus_type, val)         \
  BUS_DEFINE_PROPERTY_GET(function, bus_type, data_type, ident)
```
?

Reply:
Hmm. I have no idea how to define `BUS_DEFINE_PROPERTY_GET_GLOBAL()` by `BUS_DEFINE_PROPERTY_GET()`...

Reply:
Oh, OK. Yeah, I missed the fact that they take different args. Looks good now.